boards/arduino-mega2560/src/lib.rs

```diff
@@ -13,7 +13,10 @@ pub use crate::pac::Peripherals;
 mod pins;
 pub use crate::pins::*;
 
-pub use crate::hal::prelude;
+pub mod prelude {
+    pub use crate::hal::prelude::*;
+    pub use crate::hal::usart::BaudrateExt as _;
```

Comment:
Also need to check whether this is affected by the special implementation.  @couchand, maybe you can give a try of this PR at 57600 baud?

Reply:
In worst case I think will be able to test this in next week.

Reply:
For mega2560 same results. I will check now Uno rev3 for same baunds

baund| UBRR|tested|
-- | -- |--|
300 | 767 |ok |
600 | 383 |ok|
1200 | 191 |ok|
2400 | 95 |ok |
4800 | 47 |ok |
9600 | 23 |ok  |
14400 | 15 |failed |
19200 | 11 |ok |
28800 | 7 |failed |
38400 | 5 |ok |
57600 | 3 | ok|
76800 | 2 |failed |
115200 | 1|ok|

Reply:
This is with the `BaudrateExt` trait, right?  So it seems 57600 and 115200 are working fine with the default implementation which means we don't need to switch for this board!

Reply:
Yes. Looks like works fine. I'm able to test today uno rev3 too. Then I will need to reupload bootloader for mega2560...

Reply:
If you don't have any objections, I'm content with the results you provided and would go ahead with merging your PR now.  Thank you so much for you work here!  You were a lot more thorough than most people would be which I appreciate a lot. :)